docs/_ext/videos.py

```diff
@@ -24,98 +24,53 @@ def generate_video_content(session, year, region, session_idx):
     data['session_idx'] = session_idx
-    return u''':template: 2017/video-details.html
-
-{title}
+    return u'''{title}
 {title_marker}
 
-{{% set talk = conferences[{year}]['{region}']['speakers'][{session_idx}] %}}
-{{% set output %}}
-{{% include conf_py_root + '/_templates/video-details-body.html' %}}
-{{% endset %}}
-.. raw:: html
+.. datatemplate::
+   :source: /_data/{year}.{region}.speakers.yaml
+   :template: videos/video-detail.html
+   :key: {session_idx}
 
-    {{{{ output|indent(4) }}}}
 '''.format(**data)
 
 
-def generate_videolisting_by_year(year):
-    return u''':template: 2017/conference-videos-archive.html
-
-Videos of {year}
-==============
-
-{{% set output %}}
-{{% set path_to_root = '../../..' %}}
-{{% set talks = sessions_by_year[{year}] %}}
-{{% include conf_py_root + "/_templates/video-listing.html" %}}
-{{% endset %}}
-
-.. raw:: html
-   
-   {{{{ output|indent(3) }}}}
-
-'''.format(year=year)
-
-
-def generate_videolisting_by_series(series):
-    return u''':template: 2017/conference-videos-archive.html
+def generate_video_listing(year, series):
+    return u'''Videos of Write the Docs {series_title} {year}
+=============================================================
 
-Videos from Write the Docs {series_title}
-=============================================
+.. toctree::
+   :glob:
+   :hidden:
 
-{{% set output %}}
-{{% set talks = sessions_by_series['{series}'] %}}
-{{% set path_to_root = '../../..' %}}
-{{% include conf_py_root + "/_templates/video-listing.html" %}}
-{{% endset %}}
+   * 
 
-.. raw:: html
-   
-   {{{{ output|indent(3) }}}}
+.. datatemplate::
+   :source: /_data/{year}.{series}.speakers.yaml
+   :template: videos/video-listing.html
+'''.format(year=year, series=series, series_title=series.upper())
 
-'''.format(series=series, series_title=series.upper())
 
 
 def main():
     conference_data = load_conference_data()
-    sessions_by_year = {}
-    sessions_by_series = {}
-    shutil.rmtree(os.path.join('videos', 'by-year'), ignore_errors=True)
-    shutil.rmtree(os.path.join('videos', 'by-series'), ignore_errors=True)
-    os.makedirs(os.path.join('videos', 'by-year'))
-    os.makedirs(os.path.join('videos', 'by-series'))
 
     for year, regions in conference_data.items():
-        year_path = os.path.join('videos', 'by-year', unicode(year))
-        sessions_by_year[year] = []
-        with io.open(year_path + '.rst', 'w') as fp:
-            print('Generated {}.rst'.format(year_path + '.rst'))
-            fp.write(generate_videolisting_by_year(year))
         for region, data in regions.items():
-            if region not in sessions_by_series:
-                sessions_by_series[region] = []
-            index_path = os.path.join('conf', region, unicode(year), 'videos')
-            shutil.rmtree(index_path, ignore_errors=True)
-            os.makedirs(index_path)
+            index_path = os.path.join('videos', region, unicode(year))
+            if not os.path.exists(index_path):
```

Comment:
Add a flag in conf config to prevent publishing of video pages by default.

Reply:
👍